libp2p/autotls/acme/api.nim

```diff
@@ -1,509 +1,519 @@
-import options, sequtils, strutils, json, uri
```

Comment:
curious if what's the better way to do this.

- could we leave types and types definitions but just remove `withAutotls` method in switch - then we technically disabled the feature?
- adding `when defined(libp2p_autotls_support):` in front of whole file removes line history (sometimes vary interesting info)

Reply:
the long-term plan is to integrate `-d` support with [nimble feature flags](https://github.com/nim-lang/nimble/pull/1346) using some suitable convention that has yet to be set in stone - it might be worth to experiment with newer nimble versions to understand what this could look like.

Reply:
> could we leave types and types definitions but just remove withAutotls method in switch - then we technically disabled the feature?

I can try this.

>  it might be worth to experiment with newer nimble versions.

This sounds good, but lets do it on a separate PR whose scope is experimenting with that feature, while getting this PR merged ASAP to fix the error found by @jakubgs